low-level/solution/lr2/server/manifest.py

```diff
@@ -0,0 +1,351 @@
+        """Get server hw resource information."""
+        server = {}
+        if resource == "hw" and "hw" in server_hw_data:
+            server.update({"hw": {}})
+            # Sorting output dictionary according to data priority.
+            for server_type in self.server_resources["hw"].keys():
+                if server_type in server_hw_data["hw"]:
+                    server["hw"][server_type] = server_hw_data["hw"][server_type]
+        else:
+            server = server_hw_data["hw"].get(resource, [])
+        return server
+
+    def set_lshw_input_data(self):
+        """
+        KvStoreFactory can not accept a dictionary as direct input and output
+        It will support only JSON, YAML, TOML, INI, PROPERTIES files. So here
+        we are fetching the lshw data and adding that to a file for further
+        execution.
+        """
+        input_file = None
+        output_file = None
+        response, err, returncode = SimpleProcess("lshw -json").run()
+        if returncode:
+            msg = f"Failed to capture Node support data. Error:{str(err)}"
+            logger.error(self.log.svc_log(msg))
+            raise ResourceMapError(errno.EINVAL, msg)
+        try:
+            with open(LSHW_FILE, 'w+') as fp:
+                json.dump(json.loads(response.decode("utf-8")), fp,  indent=4)
+            with open(MANIFEST_OUTPUT_FILE, 'w+') as fp:
+                json.dump({}, fp,  indent=4)
+            input_file = KvStoreFactory.get_instance(f'json://{LSHW_FILE}').load()
+            output_file = KvStoreFactory.get_instance(f'json://{MANIFEST_OUTPUT_FILE}')
+        except Exception as e:
+            msg = "Error in getting {0} file: {1}".format(LSHW_FILE, e)
+            logger.error(self.log.svc_log(msg))
+            raise ResourceMapError(errno.EINVAL, msg)
+        return input_file, output_file
+
+    def map_manifest_server_data(self, field, manifest_key, data, kv_key):
+        """Mapping actual lshw output data with standard structured manifest data."""
+        parent_id = ""
+        base_key = '>'.join(kv_key.split('>')[:-1])
+        if base_key:
+            value = data.get(base_key+'>'+field)
+        else:
+            value = data.get(field)
+        value = value.replace(" (To be filled by O.E.M.)", "") \
+            if value else 'NA'
+        if field == 'id' and '>' in kv_key:
+            parent_key = '>'.join(kv_key.split('>')[:-2])
+            field = '>'+field if parent_key else field
+            parent_id = data.get(parent_key+field)+"-"
+        self.kv_dict[manifest_key] = parent_id + value
+
+    def get_manifest_output_data(self):
+        """Returns JSON data in the manifest output file."""
+        data = {}
+        try:
+            with open(MANIFEST_OUTPUT_FILE) as json_file:
+                data = json.loads(json_file.read())
+        except Exception as e:
+            msg = "Error in getting {0} file: {1}".format(json_file, e)
+            logger.error(self.log.svc_log(msg))
```

Comment:
In line 266, I think it should be `MANIFEST_OUTPUT_FILE` in place of `json_file` in msg because in case of `exception`, `json_file` variable will not be defined.

Reply:
Done